programs/marinade-sdk/src/checks.rs

```diff
@@ -123,7 +126,7 @@ pub fn check_token_owner(token: &TokenAccount, owner: &Pubkey, field_name: &str)
 // check that the account is delegated and to the right validator
 // also that the stake amount is updated
 pub fn check_stake_amount_and_validator(
-    stake_state: &StakeState,
```

Comment:
`StakeState` and `StakeStateV2` is fully backward compatible? Can you check whether marinade official stake account is fully 100% compatible with `StakeStateV2` from solana?

Reply:
I think it fully backward compatible. Because `StakeStateV2` just added `StakeFlags`. Also keep the same stake account and same implementation of StakeState.
You can check [StateStake](https://github.com/solana-program/stake/blob/main/interface/src/state.rs#L93) vs [StateStakeV2](https://github.com/solana-program/stake/blob/main/interface/src/state.rs#L152)